plugin/keyring_vault/vault_credentials_parser.h

```diff
@@ -23,21 +23,20 @@ namespace keyring
       optional_value.insert("vault_ca");
     }
 
-    my_bool parse(std::string *file_url, Vault_credentials *vault_credentials);
+    bool parse(const std::string &file_url, Vault_credentials *vault_credentials);
```

Comment:
I would prefer OO programming instead of global function. Maybe one day I will want to extend this call or create a mock of it.